### PR TITLE
Support Rails edge's custom deprecators

### PR DESCRIPTION
**Pull request by camertron**
### What are you trying to accomplish?

We're seeing some tests fail in dotcom CI because Rails edge has deprecated the `ActiveSupport::Deprecation` singleton in favor of per-library deprecators. This PR adds a PVC deprecator when riding the edge.

### Integration

No changes necessary in production.

#### Risk Assessment
  

- [x] **Low risk** the change is small, highly observable, and easily rolled back.
- [ ] **Medium risk** changes that are isolated, reduced in scope or could impact few users. The change will not impact library availability.
- [ ] **High risk** changes are those that could impact customers and SLOs, low or no test coverage, low observability, or slow to rollback.

### Accessibility

- **No new axe scan violation** - This change does not introduce any new [axe scan](https://thehub.github.com/epd/engineering/dev-practicals/frontend/accessibility/readiness-routine/development/#axe-scans) violations.

### CODE REVIEW

File: app/components/primer/beta/button.rb [hunk 1]
```diff
@@ -81,7 +81,7 @@ class Button < Primer::Component
       renders_one :tooltip, lambda { |**system_arguments|
         raise ArgumentError, "Buttons with a tooltip must have a unique `id` set on the `Button`." if @id.blank? && !Rails.env.production?
 
-        ActiveSupport::Deprecation.warn("Buttons with visible text should not use a `label` tooltip. Consider using Primer::Beta::IconButton instead.") if system_arguments[:type] == :label
+        deprecation.warn("Buttons with visible text should not use a `label` tooltip. Consider using Primer::Beta::IconButton instead.") if system_arguments[:type] == :label
```
Comment (jonrohan) [inline app/components/primer/beta/button.rb:84]: Should this one be `::Primer::ViewComponents.`?

File: app/components/primer/component.rb [hunk 2]
```diff
@@ -141,5 +141,9 @@ def should_raise_error?
     def should_raise_aria_error?
       !Rails.env.production? && raise_on_invalid_aria? && !ENV["PRIMER_WARNINGS_DISABLED"]
     end
+
+    def deprecation
+      ::Primer::ViewComponents.deprecation
+    end
```
Comment (jonrohan) [inline app/components/primer/component.rb:147]: Why not call `::Primer::ViewComponents.` directly above?

File: app/components/primer/beta/button.rb (hunk 1 above)
Comment (camertron) [reply app/components/primer/beta/button.rb:84]: You can call `deprecation` in anything that inherits from `Primer::Component` 👍

File: app/components/primer/component.rb (hunk 2 above)
Comment (jonrohan) [reply app/components/primer/component.rb:147]: Unless you plan to do any extra work in the deprecation call, it seems like an unnecessary abstraction

**Review by jonrohan — APPROVED** [review]
👯 👍🏻

**Pull request merged**